File: apps/mobile/lib/domain/entities/review.dart
class Review {
  final String id;
  final String productId;
  final int rating;
  final String? title;
  final String? comment;
  final String? reviewerName;
  final String createdAt;

  Review({
    required this.id,
    required this.productId,
    required this.rating,
    this.title,
    this.comment,
    this.reviewerName,
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    final profile = json['profiles'] as Map<String, dynamic>?;
    return Review(
      id: json['id'] as String,
      productId: json['product_id'] as String,
      rating: json['rating'] as int? ?? 0,
      title: json['title'] as String?,
      comment: json['comment'] as String?,
      reviewerName: profile?['full_name'] as String?,
      createdAt: json['created_at'] as String,
    );
  }
}

File: packages/models/src/review.ts
import type { Timestamp } from './common';

export interface Review {
  id: string;
  product_id: string;
  order_id: string;
  user_id: string;
  rating: number;
  title?: string;
  comment: string;
  images?: string[];
  is_verified_purchase: boolean;
  is_published: boolean;
  helpful_count: number;
  created_at: Timestamp;
  updated_at: Timestamp;
}

export interface Rating {
  id: string;
  product_id: string;
  average_rating: number;
  rating_1_count: number;
  rating_2_count: number;
  rating_3_count: number;
  rating_4_count: number;
  rating_5_count: number;
  total_count: number;
  updated_at: Timestamp;
}
